bugbug/tools/code_review.py

```diff
@@ -117,9 +138,145 @@ def get_review_request_by_id(self, review_id: int) -> ReviewRequest:
+                if hunk.target_start <= comment.start_line < target_end(hunk)
+                or hunk.source_start <= comment.start_line < source_end(hunk)
+            ]
+
+            # If there is more than one matching hunk, choose the one where the
+            # line number of the comment corresponds to an added or deleted line. We
+            # prioritize added lines over deleted lines because comments are more
+            # likely to be on added lines than deleted lines.
+            if len(matching_hunks) > 1:
+                for hunk in matching_hunks:
+                    for line in hunk:
+                        if line.is_added and line.target_line_no == comment.start_line:
+                            return hunk
+
+                    for line in hunk:
+                        if (
+                            line.is_removed
+                            and line.source_line_no == comment.start_line
+                        ):
+                            return hunk
+
+            if len(matching_hunks) != 0:
+                return matching_hunks[0]
+
+        elif comment.on_removed_code:
+            for hunk in patched_file:
+                if hunk.source_start <= comment.start_line < source_end(hunk):
+                    return hunk
+
+        else:
+            for hunk in patched_file:
+                if hunk.target_start <= comment.start_line < target_end(hunk):
+                    return hunk
+
+    def retrieve_comments_with_hunks(self):
+        def comment_filter(comment: InlineComment):
+            comment_content = comment.content
+
+            # Ignore very short and very log comments
+            if not 50 < len(comment_content) < 500:
+                return False
+
+            # Ignore comments with URLs
+            if "https://" in comment_content or "http://" in comment_content:
+                return False
+
+            #  Ignore nit comments
+            if self.NIT_PATTERN.search(comment_content):
+                return False
+
+            # Ignore comments with code blocks
+            if "```" in comment_content:
+                return False
+
+            comment_lower = comment_content.lower()
+            if any(
+                phrase in comment_lower
+                for phrase in [
+                    "wdyt?",
+                    "what do you think?",
+                    "you explain",
+                    "understand",
+                ]
+            ):
+                return False
+
+            return True
+
+        for diff_id, comments in self.get_all_inline_comments(comment_filter):
+            try:
+                patch_set = PatchSet.from_string(
+                    self.load_patch_by_id(diff_id).raw_diff
+                )
+            except UnidiffParseError:
+                # TODO: use log instead of print
+                print(f"Failed to parse {diff_id}")
+                continue
+
+            file_map = {
+                patched_file.path: patched_file
```

Comment:
IIRC there might be cases where the path starts with "a/" or "b/" and it should be removed (similar to https://github.com/michael-yx-wu/environment/commit/59f9fd17ea9b41c0730dd0e5ae236aef27149248.diff)

Reply:
That is correct when using `source_file` or `target_file`, but the getter method for the `path` property already strips that: https://github.com/matiasb/python-unidiff/blob/0f1062137b3c8d2b6b73cad3508dfb8862e66b87/unidiff/patch.py#L400-L401